### PR TITLE
chore: move `LearningRateExp` to `deepmd.utils.learning_rate`

### PR DESCRIPTION
**Pull request by njzjz**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new exponential decay learning rate scheduler to enhance training efficiency.
	- Added functionality to compute learning rates at specific training steps.

- **Bug Fixes**
	- Removed the outdated `LearningRateExp` class from the previous module to avoid confusion.

### CODE REVIEW

File: deepmd/utils/learning_rate.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        if decay_rate is not None:
+            self.decay_rate = decay_rate
+        self.min_lr = stop_lr
+
+    def value(self, step) -> np.ndarray:
+        """Get the learning rate at the given step."""
+        step_lr = self.start_lr * np.power(self.decay_rate, step // self.decay_steps)
+        if step_lr < self.min_lr:
+            step_lr = self.min_lr
+        return step_lr
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/utils/learning_rate.py:53]:
```suggestion
    def value(self, step) -> np.float64:
        """Get the learning rate at the given step."""
        step_lr = self.start_lr * np.power(self.decay_rate, step // self.decay_steps)
        if step_lr < self.min_lr:
            step_lr = self.min_lr
        return step_lr
```

**Comment by njzjz** [comment]
Or, should we move it to `deepmd.dpmodel.utils.learning_rate`?

**Comment by iProzd** [comment]
> Or, should we move it to `deepmd.dpmodel.utils.learning_rate`?

Is it easy to merge both tf and pt? If not (so far), `deepmd.dpmodel.utils.learning_rate` is better to me.

**Review by wanghan-iapcm — APPROVED** [review]

**Comment by njzjz** [comment]
> Is it easy to merge both tf and pt?

No, for performance, TF needs a static graph, even for the learning rate.

**Review by iProzd — APPROVED** [review]
